naz/client.py

```diff
@@ -1852,6 +1852,12 @@ def _msg_to_log(self, msg: bytes) -> str:
                 # `client.reader` and `client.writer` should not have timeouts since they are non-blocking
                 # https://github.com/komuw/naz/issues/116
                 command_length_header_data = await self.reader.read(4)
+
+                # make sure the header data be 4 bytes
```

Comment:
@jwenjian I wonder if using `await self.reader.readexactly(4)`[1] would be better? 
although that raises `IncompleteReadError`[2] which we would have to handle somehow. 

1. https://docs.python.org/3.6/library/asyncio-stream.html#asyncio.StreamReader.readexactly
2. https://docs.python.org/3.6/library/asyncio-stream.html#asyncio.IncompleteReadError

Reply:
Also relevant[3].

I do not know which is better. Let me know your thoughts

3. https://stackoverflow.com/questions/28452724/method-asyncio-streams-streamreader-readbyte-count-reads-too-few-bytes

Reply:
The `readexactly` method is better if we have a proper way to handle the `IncompleteReadError`

log an error and ignore it or raise it?

Reply:
It seems there is no `timeout` in the `read*()` functions? 

So if the `readexactly(4)`  raises an error, it means that there is no more data and the read partial bytes(length must be less than 4) is not part of an valid SMPP PDU, so I suggest to abandon it then continue the while loop.

Reply:
In my opinion, like the way you handle the message body in [Line 1900](https://github.com/komuw/naz/blob/master/naz/client.py#L1900), it should works fine to the headers.

also don't know which is better, so maybe we can use the way I do in this pr util we find out a more efficient way.

Reply:
thanks for your input. I think we can start with this and improve if/when it becomes neccessary

I'll think about it some more.